### ISSUE TITLE
每次一更新就挂掉

### ISSUE THREAD

**Issue by kemchenj** [XX-net/XX-Net#688]
每次更新完就 “状态页显示空白，很可能GAEProxy启动失败” ，不知道是不是跟我启动了开机自启动有关（在xx-net里面设置自启的），每次我都是把更新的最新版的压缩包从data里面拷出来，把旧版删了，然后把新版的压缩包解压出来重新配置一遍就好了，更新的方式有问题？
PS：win10 10240 th1

**Comment by xxnet** [comment]
你原来是什么版本？



在 2015年7月21日 上午11:19，Jie Chen <notifications@github.com>写道：

**Comment by kemchenj** [comment]
2.01升2.04
之前1.83（不确定）升2.01也是一样

**Comment by kemchenj** [comment]
2.01升2.04
之前升到2.01的时候也是一样，好几次了

**Comment by mochen91** [comment]
@xxnet  是的 感觉自动更新有点鸡肋 ，从来没用到过 ，当每次需要更新的时候XXNET就不能用了，翻不了墙，然后检查更新也检查不出来了。今天也遇到了这种情况“状态页显示空白，很可能GAEProxy启动失败”” 作为小白就不知道该怎么办了，看到项目页有版本更新，就下载2.0.5（使用迅雷下载的），结果解压时说文件不完整。

**Comment by xxnet** [comment]
恩，自动更新应该抛开翻墙功能，目前直接通过github，不能翻墙也还能升级，这个可以改掉。
升级可以考虑做成独立的模块，可以先停止xx-net的执行，然后再启动升级。

本来“重启自身” 这件事情就不好搞，还要先覆盖自己，然后再重启自己，呵呵。

配置的保留，本来就是希望能够维持原来的设置，才重点开发了自动升级。
  不过这次1.x.x升级 2.x.x，本身就是结构调整，应该提示用户重新下载会更好，呵呵。

在 2015年7月22日 下午12:06，mochen91 <notifications@github.com>写道：

**Comment by mochen91** [comment]
原来如此~

在 15/7/22，xxnet<notifications@github.com> 写道：

**Comment by kemchenj** [comment]
@xxnet 2.05升级没有出现问题了。。谢谢你一直维护这个计划。。
